### PR TITLE
Bump cosmos to v0.46.8-pio-2 with State Listening support

### PR DESCRIPTION
**Pull request by egaxhaj**
## Description

- bumps Cosmos SDK to v0.46.8-pio-2
- adds State Listening support
- updates third_party proto files from v0.46.8-pio-2

closes: #1332

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by SpicyLemon — APPROVED** [review]

**Review by nullpointer0x00 — CHANGES_REQUESTED** [review]
Everything looks good.  Just found one issue with a comment being incorrect.  Once corrected I'm ready to approve.

File: third_party/proto/cosmos/streaming/abci/v1/grpc.proto [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// ListenCommitRequest is the request type for the ListenCommit RPC method
+message ListenCommitRequest {
+  // explicitly pass in block height as ResponseCommit does not contain this info
+  int64                          block_height               = 1;
+  tendermint.abci.ResponseCommit res                        = 2;
+  repeated cosmos.base.store.v1beta1.StoreKVPair change_set = 3;
+}
+
+// ListenBeginBlockResponse is the response type for the ListenCommit RPC method
+message ListenCommitResponse {}
```
[the remark this answers is not in the document]
Comment (SpicyLemon) [reply third_party/proto/cosmos/streaming/abci/v1/grpc.proto:48]: Oops.

This is in a third-party proto. It can't be updated here.

**Review by nullpointer0x00 — APPROVED** [review]

File: third_party/proto/cosmos/streaming/abci/v1/grpc.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (egaxhaj) [reply third_party/proto/cosmos/streaming/abci/v1/grpc.proto:48]: I'll make a note to update this upstream. Thanks for catching it.

**Review by SpicyLemon — APPROVED** [review]

**Review by derekadams — APPROVED** [review]
LGTM

**Review by nullpointer0x00 — APPROVED** [review]

**Pull request merged**